src/test/java/edu/cornell/kfs/sys/service/impl/CuUserFavoriteAccountServiceImplTest.java

```diff
@@ -1,105 +1,136 @@
+        final GeneralLedgerPendingEntrySourceDetail acct = userFavoriteAccountService.getPopulatedNewAccount(favoriteAccount, accountClass);
+        Assert.assertNotNull("Account should be populated", acct);
+        Assert.assertTrue("Account should be " + accountType + " Account", accountClass.isInstance(acct));
+        Assert.assertEquals("Account Number should be populated", favoriteAccount.getAccountNumber(), acct.getAccountNumber());
+        Assert.assertEquals("Object Code should be populated", favoriteAccount.getFinancialObjectCode(), acct.getFinancialObjectCode());
+
+        if (acct instanceof IWantAccount) {
+            final KualiDecimal accountLinePercent = ((IWantAccount) acct).getAmountOrPercent();
+            Assert.assertEquals("Incorrect percentage (comparison against 100% should have been zero)", 0, accountLinePercent.compareTo(new KualiDecimal(100)));
+            Assert.assertEquals("Amount-or-Percent indicator should be Percent", CUPurapConstants.PERCENT, ((IWantAccount) acct).getUseAmountOrPercent());
+        } else {
+            final BigDecimal accountLinePercent = ((PurApAccountingLineBase) acct).getAccountLinePercent();
+            Assert.assertEquals("Incorrect percentage (comparison against 100% should have been zero)", 0, accountLinePercent.compareTo(new BigDecimal(100)));
+        }
     }
-    
-    /*
-     * test retrieve favorite account based on the selected favorite account PK
-     */
+
+    @Test
+    public void testPopulatedNewAccountNull() {
+        GeneralLedgerPendingEntrySourceDetail acct = userFavoriteAccountService.getPopulatedNewAccount(null, PurchaseOrderAccount.class);
+        Assert.assertNull("Account should not be populated", acct);
+    }
+
+    @Test
     public void testSelectedFavoriteAccount() {
-        FavoriteAccount favoriteAccount1 = ((List<FavoriteAccount>)SpringContext.getBean(BusinessObjectService.class).findAll(FavoriteAccount.class)).get(0);
-        
+        FavoriteAccount favoriteAccount1 = FavoriteAccountFixture.FAVORITE_ACCOUNT_1.createFavoriteAccount();
         FavoriteAccount favoriteAccount2 = userFavoriteAccountService.getSelectedFavoriteAccount(favoriteAccount1.getAccountLineIdentifier());
-        assertTrue("Favorite Account should exist", favoriteAccount2 != null);
-        assertTrue("should have the same PK", favoriteAccount1.getAccountLineIdentifier().equals(favoriteAccount2.getAccountLineIdentifier()));
+        Assert.assertTrue("Favorite Account should exist", favoriteAccount2 != null);
+        Assert.assertTrue("should have the same PK", favoriteAccount1.getAccountLineIdentifier().equals(favoriteAccount2.getAccountLineIdentifier()));
+    }
+
+    private class MockBusinessObjectServiceImpl extends BusinessObjectServiceImpl {
+
+        @Override
+        public <T extends BusinessObject> T findByPrimaryKey(Class<T> clazz, Map<String, ?> primaryKeys) {
+            return (T)FavoriteAccountFixture.FAVORITE_ACCOUNT_1.createFavoriteAccount();
```

Comment:
In that case, this should be fine unless we anticipate needing to add more test cases that need a different return value.